src/ppx/scene/scene_material.cpp

```diff
@@ -0,0 +1,217 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ppx/scene/scene_material.h"
+#include "ppx/grfx/grfx_device.h"
+
+namespace ppx {
+namespace scene {
+
+// -------------------------------------------------------------------------------------------------
+// Image
+// -------------------------------------------------------------------------------------------------
+Image::Image(
+    grfx::Image*            pImage,
+    grfx::SampledImageView* pImageView)
+    : mImage(pImage),
+      mImageView(pImageView)
+{
+}
+
+Image::~Image()
+{
+    if (mImageView) {
+        auto pDevice = mImageView->GetDevice();
+        pDevice->DestroySampledImageView(mImageView);
+    }
+
+    if (mImage) {
+        auto pDevice = mImage->GetDevice();
+        pDevice->DestroyImage(mImage);
+    }
+}
+
+// -------------------------------------------------------------------------------------------------
+// Sampler
+// -------------------------------------------------------------------------------------------------
+Sampler::Sampler(
+    grfx::Sampler* pSampler)
+    : mSampler(pSampler)
+{
+}
+
+Sampler::~Sampler()
+{
+    if (mSampler) {
+        auto pDevice = mSampler->GetDevice();
+        pDevice->DestroySampler(mSampler);
+        mSampler.Reset();
+    }
+}
+
+// -------------------------------------------------------------------------------------------------
+// Texture
+// -------------------------------------------------------------------------------------------------
+Texture::Texture(
+    const scene::ImageRef   image,
+    const scene::SamplerRef sampler)
+    : mImage(image),
+      mSampler(sampler)
+{
+}
+
+Texture::~Texture()
```

Comment:
There is no overridden function, those virtual dtors are not mandatory AFAIK.
And if you need one, what about using `virtual ~Image() = default;` since those are empty?

Reply:
This is a left over paranoia from when certain compilers/linkers had a lot of issue with classes without a fully specified destructor.

Reply:
Removed.